extension/src/server/GradleServer.ts

```diff
@@ -29,27 +31,33 @@ export class GradleServer {
     constructor(
         private readonly opts: ServerOptions,
         private readonly context: vscode.ExtensionContext,
-        private readonly logger: Logger
+        private readonly logger: Logger,
+        private messageForwardHandler: MessageForwardHandler
     ) {}
 
     public async start(): Promise<void> {
         if (isDebuggingServer()) {
-            this.port = 8887;
             this.fireOnStart();
         } else {
-            this.port = await getPort();
+            this.gradleServerPort = await getPort();
             const cwd = this.context.asAbsolutePath("lib");
             const cmd = path.join(cwd, getGradleServerCommand());
             const env = await getGradleServerEnv();
+            const bundleDirectory = await this.getBundleDirectory();
             if (!env) {
                 await vscode.window.showErrorMessage(NO_JAVA_EXECUTABLE);
                 return;
             }
-            const args = [String(this.port)];
+            const javaExecPath = getRedHatJavaExecutablePath();
 
-            this.logger.debug("Starting server");
+            //Get the Java executable used by JDT.LS, which will be higher than JDK 17.
+            if (!javaExecPath) {
+                await vscode.window.showErrorMessage("No Red Hat Java Extension Pack Found");
+                return;
+            }
+            const serverPipeName = this.messageForwardHandler.getBuildServerPipeName();
+            const args = [String(this.gradleServerPort), serverPipeName, bundleDirectory, javaExecPath];
```

Comment:
https://github.com/microsoft/vscode-gradle/blob/144c86574922c3a7ff8395e18f041793b8127f15/extension/jdtls.ext/com.microsoft.gradle.bs.importer/src/com/microsoft/gradle/bs/importer/ImporterPlugin.java#L146-L156
These part should be equal with how we get it in Java. Maybe need to add more code to get a Java executable path that higher than JDK17.

Reply:
Two conditions here:

1. If user installs `redhat.java`
    - Check the embedded jre
    - Check setting `java.jdt.ls.java.home` & `java.home`

2. If user does not installs `redhat.java`, in this case, the build server will not be invoked. Then we can use the previous jdk search logic to find a jdk, and launch the server. In the merged server, you can check the version of jdk used to launch itself, and only start the build server when it's higher than 17